app/models/landing_page/block/statistics.rb

```diff
@@ -42,7 +42,14 @@ def opened_csv
     end
 
     def csv_from_url
-      CSV.parse(URI.parse(attachment.url).open, headers: true)
+      # TODO: asset_id is a 24 char hexadecimal number which doesn't appear elsewhere in the attachment hash.
+      #       We should get Whitehall to provide it rather than parsing the URL.
+      match = attachment.url.match(%r{/media/(?<asset_id>[[:xdigit:]]{24})/(?<filename>[^/]+[.]csv)\z})
+      raise "Unexpected URL format #{attachment.url} - cannot load CSV" unless match
```

Comment:
If an error is raised does that mean that the page will fail?

At the moment the code in main has the chart fails silently. The page still loads, but omits the chart. Will that still happen here?

This PR is missing the most up to date code. The PR needs to be rebased to make sure that the tests still pass.

Reply:
Tests still work after a rebase, it looks like.

I'm in two minds about the chart omission. On the one hand, it's nice and safe. On the other hand, it leads to a bit of a difficult to diagnose problem (if a chart is missing, what's causing that? missing attachment? broken attachment? chart block typo in YAML?) If safety is our priority, then we can catch this. Perhaps we should always catch blocks that raise an error in the BlockFactory and just safely drop them?

Reply:
We could add:
```    def opened_csv
      @opened_csv ||= attachment ? csv_from_url : csv_from_file
    rescue StandardError => e
      Rails.logger.error(e)
      nil
    end```

Reply:
Sounds good.

Reply:
The main thing is that the rest of the page loads and doesn't bomb.

Reply:
Looks like Keith's resolved this 👇🏻